src/SmartComponents/SubscriptionsUtilized/SubscriptionsUtilizedHelpers.js

```diff
@@ -0,0 +1,97 @@
+import moment from 'moment';
+import { RHSM_API_RESPONSE_DATA_TYPES } from './Constants';
+
+/**
+ * Generate a range of dates.
+ *
+ * @param {Date} date
+ * @param {number} subtract
+ * @param {string} measurement
+ * @returns {{endDate: Date, startDate: Date}}
+ */
+export const setRangedDateTime = (date = new Date(), subtract = 1, measurement = 'days') => ({
+    startDate: moment
+    .utc(date)
+    .startOf(measurement)
+    .subtract(subtract, measurement)
+    .toDate(),
+    endDate: moment
+    .utc(date)
+    .startOf(measurement)
+    .endOf('days')
+    .toDate()
+});
+
+/**
+ * Apply a set of schemas using either an array of objects in the
+ * form of [{ madeUpKey: 'some_api_key' }], or an array of arrays
+ * in the form of [['some_api_key','another_api_key']]
+ *
+ * @param {Array} schemas
+ * @param {*} initialValue
+ * @returns {Array}
+ */
+export const setResponseSchemas = (schemas = [], initialValue) =>
+    schemas.map(schema => {
+        const generated = {};
+        const arr = (Array.isArray(schema) && schema) || Object.values(schema);
+        arr.forEach(value => {
+            generated[value] = initialValue;
+        });
+        return generated;
+    });
+
+/**
+ * Filter report and capacity data against expected API response. Apply percentage and date.
+ *
+ * @param {Array} report
+ * @param {Array} capacity
+ * @param {string} filter
+ * @return {({}|{date: Date, report: (number|null|undefined), capacity: (number|null|undefined),
+ *     percentage: (number|null|undefined)})}
+ */
+export const filterChartData = (report = [], capacity = [], filter) => {
+    const reportData = report.reverse();
+    const capacityData = capacity.reverse();
+    let chartData = {};
+
+    const [responseSchema = {}] = setResponseSchemas([
+        RHSM_API_RESPONSE_DATA_TYPES
+    ]);
+
+    for (let index = 0; index < reportData.length; index++) {
```

Comment:
this could be rewritten as a map, with a trigger on reportData being true (having a length)

```
reportData.length && reportData.map((value, index) => {
```
following line, (const value)  could be removed cuz it'll be value from above

Reply:
This one I'll side step. 

We can use `find` and avoid the mega loop all together since we only need one record along with the index to match the parallel API call.

Reasons
- the API was flaking around duplicate records... appears fixed, but as a fallback
- `filterChartData` function is temporary until the API provides an updated end point
- continue and break vs ifs and returns works out about the same